alacritty/src/input.rs

```diff
@@ -710,7 +710,15 @@ impl<'a, T: EventListener, A: ActionContext<T>> Processor<'a, T, A> {
 
     /// Process a received character.
     pub fn received_char(&mut self, c: char) {
-        if *self.ctx.suppress_chars() || self.ctx.terminal().mode().contains(TermMode::VI) {
+        if *self.ctx.suppress_chars() {
+            return;
+        }
+
+        if self.ctx.config().ui_config.mouse.hide_when_typing {
+            self.ctx.window_mut().set_mouse_visible(false);
+        }
+
+        if self.ctx.terminal().mode().contains(TermMode::VI) {
```

Comment:
I'm not a huge fan of this approach, it's kinda messy. 

I think it would be better to explicitly hide the mouse courser when the `ViMotion` command is triggered.

Reply:
This here exists for keys pressed that are *not* bound to an action. If an action is triggered `suppress_chars` is `true`. But if there is no action and we're in vi mode the cursor gets hidden, if that behaviour is not desired, I can "merge" the `suppress_chars` and vi mode check back together.

Reply:
I think we should, it'll also simplify Vi mode cursor hide handling.

Reply:
Merged